spec/models/spree/variant_spec.rb

```diff
@@ -589,10 +589,14 @@ module Spree
         it "returns options_text if display_as is blank" do
           v = build_stubbed(:variant)
           v1 = build_stubbed(:variant, display_as: "")
+          p = create(:simple_product, variant_unit_name: "items")
+          v2 = build_stubbed(:variant, product: p)
           allow(v).to receive(:options_text).and_return "ponies"
           allow(v1).to receive(:options_text).and_return "ponies"
+          allow(v2).to receive(:options_text).and_return "ponies"
           expect(v.unit_to_display).to eq("ponies")
           expect(v1.unit_to_display).to eq("ponies")
+          expect(v2.unit_to_display).to eq("ponies")
```

Comment:
This example seems unrelated to your code change. Your added code above handles `variant_unit_name` and `variant_unit_scale` which is not tested by this spec.

When you address a bug, it's best to write a spec that failing with the buggy code and then passes with your fixed code. A nice introduction to Test Driven Development is an old RailsCast episode: http://railscasts.com/episodes/275-how-i-test

I think that you chose the right spec file and roughly the right location but it would be nice to see a new example `it "displays the variant unit name if there is no unit scale" do`.